src/components/Card/EmptyStateCard.js

```diff
@@ -1,151 +1,53 @@
-    // using one of the props that apply to the default button component.
-    const anyActionParam =
-      props.actionDisabled || props.actionText || props.onActivate
-    if (prevProps.action !== props.action && props.action && anyActionParam) {
-      warn(
-        'Using `action` takes priority over `actionDisabled`, `actionText`, ' +
-          'and `onActivate`. Please use either `action` or any of these ' +
-          'props.'
-      )
-    }
-  }
-
-  // Render icon, even when provided as a component (deprecated)
-  iconElement() {
-    const { icon } = this.props
-    return !icon || React.isValidElement(icon)
-      ? icon
-      : React.createElement(icon)
-  }
-
-  // Render the action (usually a button)
-  actionElement() {
-    const {
-      action,
-      actionButton,
-      actionDisabled,
-      actionText,
-      onActivate,
-    } = this.props
-
-    // If an action node is provided, it takes priority above the other props
-    if (action) {
-      return action
-    }
-
-    // If no label is provided, the button is not displayed
-    if (!actionText) {
-      return null
-    }
-
-    // Use the provided actionButton if it exists, or a local component.
-    const Button = actionButton || DefaultActionButton
-
-    return (
-      <Button disabled={actionDisabled} mode="strong" onClick={onActivate} wide>
-        {actionText}
-      </Button>
-    )
-  }
-
-  render() {
-    const { title, text, ...props } = this.props
-    const extraProps = omit(props, Object.keys(EmptyStateCard.propTypes))
-    return (
-      <Main {...extraProps}>
-        <section>
-          {this.iconElement()}
-          <Heading>
-            <Text color={theme.accent} weight="bold" size="large">
-              {title}
-            </Text>
-          </Heading>
-          {text && <Text.Block>{text}</Text.Block>}
-          <ActionWrapper>{this.actionElement()}</ActionWrapper>
-        </section>
-      </Main>
-    )
-  }
+import { GU, textStyle } from '../../style'
+import { useTheme } from '../../theme'
+
+const EmptyStateCard = React.memo(function EmptyStateCard({
+  actionButton,
+  actionText,
+  icon,
+  onClick,
+  text,
+}) {
+  const theme = useTheme()
+
+  return (
+    <Card
+      css={`
+        display: grid;
+        grid-template-columns: 1fr;
+        grid-template-rows: auto 1fr auto;
+        padding: ${2.5 * GU}px ${2 * GU}px ${2 * GU}px;
+        text-align: center;
+      `}
+    >
+      {icon}
+      <span
+        css={`
+          color: ${theme.content};
+          ${textStyle('title4')};
+        `}
+      >
+        {text}
+      </span>
+      {actionButton && (
+        <Button onClick={onClick} wide mode="strong">
+          {actionText}
+        </Button>
+      )}
+    </Card>
+  )
+})
+
+EmptyStateCard.propTypes = {
+  actionButton: PropTypes.bool,
+  actionText: PropTypes.string,
+  icon: PropTypes.node.isRequired,
+  onClick: PropTypes.func.isRequired,
+  text: PropTypes.node,
```

Comment:
@AquiGorka we used to have a default for this, but I think we should make it required as app authors should change it anyway. What do you think?

```suggestion
  text: PropTypes.node.isRequired,
```